Colours worn at different days of the week my staffs at Bincom ICT

In [8]:
Colours_dress_by_worker_DuringTheWeek = {
    "Monday": ["GREEN", "YELLOW", "GREEN", "BROWN", "BLUE", "PINK", "BLUE", "YELLOW", "ORANGE", "CREAM", 
          "ORANGE", "RED", "WHITE", "BLUE", "WHITE", "BLUE", "BLUE", "BLUE", "GREEN"],
    
    "Tuesday": ["ARSH", "BROWN", "GREEN", "BROWN", "BLUE", "BLUE", "BLEW", "PINK", "PINK", 
           "ORANGE", "ORANGE", "RED", "WHITE", "BLUE", "WHITE", "WHITE", "BLUE", "BLUE", "BLUE"],
    
    "Wednesday": ["GREEN", "YELLOW", "GREEN", "BROWN", "BLUE", "PINK", "RED", "YELLOW", "ORANGE", 
             "RED", "ORANGE", "RED", "BLUE", "BLUE", "WHITE", "BLUE", "BLUE", "WHITE", "WHITE"],
    
    "Thursday": ["BLUE", "BLUE", "GREEN", "WHITE", "BLUE", "BROWN", "PINK", "YELLOW", "ORANGE", 
            "CREAM", "ORANGE", "RED", "WHITE", "BLUE", "WHITE", "BLUE", "BLUE", "BLUE", "GREEN"],
    
    "Friday": ["GREEN", "WHITE", "GREEN", "BROWN", "BLUE", "BLUE", "BLACK", "WHITE", 
          "ORANGE", "RED", "RED", "RED", "WHITE", "BLUE", "WHITE", "BLUE", "BLUE", "BLUE", "WHITE"]
}

In [26]:
for colours in Colours_dress_by_worker_DuringTheWeek.values():
    # To remove duplicate colours
    print(list(set(colours)))
    

['RED', 'PINK', 'GREEN', 'YELLOW', 'ORANGE', 'BROWN', 'CREAM', 'WHITE', 'BLUE']
['BLEW', 'RED', 'GREEN', 'PINK', 'ORANGE', 'BROWN', 'WHITE', 'ARSH', 'BLUE']
['RED', 'PINK', 'GREEN', 'YELLOW', 'ORANGE', 'BROWN', 'WHITE', 'BLUE']
['RED', 'PINK', 'GREEN', 'YELLOW', 'ORANGE', 'WHITE', 'BROWN', 'CREAM', 'BLUE']
['RED', 'GREEN', 'ORANGE', 'WHITE', 'BROWN', 'BLACK', 'BLUE']


In [38]:
def Worker_colour_counter(arg):
    
    """
    Function to count the number 
    of workers that put on a particular
    kind of colour
    
    """
    #initial color count
    blue = 0
    red = 0
    pink = 0
    green = 0 
    yellow = 0 
    orange = 0
    brown = 0
    cream = 0
    white = 0
    arsh = 0
    blew = 0 
    black = 0
    
    for colours in arg:
        if colours == "BLUE":
            blue += 1
        elif colours == "RED":
            red += 1
        elif colours == "PINK":
            pink += 1
        elif colours == "GREEN":
            green += 1
        elif colours == "YELLOW":
            yellow += 1
        elif colours == "ORANGE":
            orange += 1
        elif colours == "BROWN":
            brown += 1
        elif colours == "CREAM":
            cream += 1
        elif colours == "WHITE":
            white += 1
        elif colours == "ARSH":
            arsh += 1
        elif colours == "BLEW":
            blew += 1
        elif colours == "BLACK":
            black += 1
        
    print("Blue: " + str(blue))
    print("Red: " + str(red))
    print("Pink: " + str(pink))
    print("Green: " + str(green))
    print("Yellow: " + str(yellow))
    print("Orange: " + str(orange))
    print("Brown: " + str(brown))
    print("Cream: " + str(cream))
    print("White: " + str(white))
    print("Arsh: " + str(arsh))
    print("Blew: " + str(blew))
    print("Black: " + str(black))

In [41]:
Worker_colour_counter(Colours_dress_by_worker_DuringTheWeek["Monday"])
print("\n")
Worker_colour_counter(Colours_dress_by_worker_DuringTheWeek["Tuesday"])
print("\n")
Worker_colour_counter(Colours_dress_by_worker_DuringTheWeek["Wednesday"])
print("\n")
Worker_colour_counter(Colours_dress_by_worker_DuringTheWeek["Thursday"])
print("\n")
Worker_colour_counter(Colours_dress_by_worker_DuringTheWeek["Friday"])
print("\n")

Blue: 6
Red: 1
Pink: 1
Green: 3
Yellow: 2
Orange: 2
Brown: 1
Cream: 1
White: 2
Arsh: 0
Blew: 0
Black: 0


Blue: 6
Red: 1
Pink: 2
Green: 1
Yellow: 0
Orange: 2
Brown: 2
Cream: 0
White: 3
Arsh: 1
Blew: 1
Black: 0


Blue: 5
Red: 3
Pink: 1
Green: 2
Yellow: 2
Orange: 2
Brown: 1
Cream: 0
White: 3
Arsh: 0
Blew: 0
Black: 0


Blue: 7
Red: 1
Pink: 1
Green: 2
Yellow: 1
Orange: 2
Brown: 1
Cream: 1
White: 3
Arsh: 0
Blew: 0
Black: 0


Blue: 6
Red: 3
Pink: 0
Green: 2
Yellow: 0
Orange: 1
Brown: 1
Cream: 0
White: 5
Arsh: 0
Blew: 0
Black: 1




In [42]:
## General method to get data for any day of the week
day_of_the_week = input("Enter day of the week to total number of workers that wore a particluar type or colour: ")
Worker_colour_counter(Colours_dress_by_worker_DuringTheWeek[day_of_the_week])


Enter day of the week to total number of workers that wore a particluar type or colour: Monday
Blue: 6
Red: 1
Pink: 1
Green: 3
Yellow: 2
Orange: 2
Brown: 1
Cream: 1
White: 2
Arsh: 0
Blew: 0
Black: 0


In [53]:
Total_blue_worn_in_the_week = 6 + 6 + 5 + 7 + 6
print(Total_blue_worn_in_the_week)

Total_red_worn_in_the_week = 1 + 1 + 3 + 1 + 3
print(Total_red_worn_in_the_week)

Total_pink_worn_in_the_week = 1 + 2 + 1 + 1 + 0
print(Total_pink_worn_in_the_week)

Total_green_worn_in_the_week = 3 + 1 + 2 + 2 + 2
print(Total_green_worn_in_the_week)

Total_yellow_worn_in_the_week = 2 + 0 + 2 + 1 + 0
print(Total_yellow_worn_in_the_week)

Total_orange_worn_in_the_week = 2 + 2 + 2 + 2 + 1
print(Total_orange_worn_in_the_week)

Total_brown_worn_in_the_week = 1 + 2 + 1 + 1 + 1
print(Total_brown_worn_in_the_week)

Total_cream_worn_in_the_week = 1 + 0 + 0 + 1 + 0
print(Total_cream_worn_in_the_week )

Total_white_worn_in_the_week = 2 + 3 + 3 + 3 + 5
print(Total_white_worn_in_the_week)

Total_arsh_worn_in_the_week =  0 + 1 + 0 + 0 + 0
print(Total_arsh_worn_in_the_week)

Total_blew_worn_in_the_week = 0 + 1 + 0 + 0 + 0
print(Total_blew_worn_in_the_week)

Total_black_worn_in_the_week = 0 + 0 + 0 + 0 + 1
print(Total_black_worn_in_the_week)


30
9
5
10
5
9
6
2
16
1
1
1


In [64]:
week_total_colour = [Total_blue_worn_in_the_week, Total_red_worn_in_the_week, Total_green_worn_in_the_week, Total_pink_worn_in_the_week, Total_yellow_worn_in_the_week, Total_orange_worn_in_the_week, Total_cream_worn_in_the_week, Total_brown_worn_in_the_week, Total_white_worn_in_the_week, Total_arsh_worn_in_the_week, Total_black_worn_in_the_week, Total_blew_worn_in_the_week ]

print(week_total_colour)

[30, 9, 10, 5, 5, 9, 2, 6, 16, 1, 1, 1]


In [143]:
import math
import numpy as np
import pandas as pd

In [78]:
data = { "colour" : [ "blue", "red", "green", "pink", "yellow", "orange", "cream", "brown", "white", "arsh", "black", "blew"],
    "Weekly colour Total" : [Total_blue_worn_in_the_week, Total_red_worn_in_the_week, Total_green_worn_in_the_week, Total_pink_worn_in_the_week, Total_yellow_worn_in_the_week, Total_orange_worn_in_the_week, Total_cream_worn_in_the_week, Total_brown_worn_in_the_week, Total_white_worn_in_the_week, Total_arsh_worn_in_the_week, Total_black_worn_in_the_week, Total_blew_worn_in_the_week]
}

In [93]:
Workers_data = pd.DataFrame(data)
Workers_data

colour  Weekly colour Total
0     blue                   30
1      red                    9
2    green                   10
3     pink                    5
4   yellow                    5
5   orange                    9
6    cream                    2
7    brown                    6
8    white                   16
9     arsh                    1
10   black                    1
11    blew                    1

In [62]:
def mean_colour_worn(arg1):
    mean_result = np.mean(arg1)
    return mean_result

In [97]:
mean_colour_worn(week_total_colour)

7.916666666666667

Answers to question 1

* The mean for the set of colors is : 7.916666666666667 (Blue)

Answers to question 2

* Mode(Most occuring item): Blue

In [94]:
def median_colour_worn(arg1):
    mean_result = np.median(arg1)
    return mean_result

In [95]:
median_colour_worn(week_total_colour)

5.5

Answers to questions 3

Median of the set of data: 5.5

In [100]:
def varience_colour_worn(arg1):
    varience_result = np.var(arg1)
    return varience_result

In [101]:
varience_colour_worn(week_total_colour)

63.243055555555564

Answers to questions 4

varience of the set of data: 63.243055555555564

In [114]:
total = Total_blue_worn_in_the_week + Total_red_worn_in_the_week + Total_green_worn_in_the_week + Total_pink_worn_in_the_week + Total_yellow_worn_in_the_week + Total_orange_worn_in_the_week + Total_cream_worn_in_the_week + Total_brown_worn_in_the_week + Total_white_worn_in_the_week + Total_arsh_worn_in_the_week + Total_black_worn_in_the_week + Total_blew_worn_in_the_week 

In [115]:
print(total)

95


In [117]:
probability_red = Total_red_worn_in_the_week / total
print(probability_red)

0.09473684210526316


Answers to questions 5

Prob(red) = 0.09473684210526316

In [223]:
import random

In [219]:
def decimal_generator(number_range):
    """
    Function to generate 
    random n digits binary 
    numbers
    
    """
    binary = ""
    for i in range(number_range):
        count = str(random.randint(0,1))
        binary += count
    return binary

In [220]:
# 4 is used since we want 4 digits binary numbers

bin_result = decimal_generator(4)
print(bin_result)

0111


In [221]:
def binary_to_decimal(decimal_num):
    decimal = int(bin_result ,2)
    print(decimal)

In [222]:
binary_to_decimal(bin_result)

7
